### PR TITLE
Upload GH action built docs as Artifacts

### PR DESCRIPTION
**Pull request by harshil21**
Uploads the built docs from the workflow run so it is easy to see the rendered html pages without building the docs again locally.

Docs on artifacts: https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts

How to view the artifact? Click on the "Details" of Test Doc Build -> Summary (left panel) -> scroll down and download the artifact

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -30,3 +30,12 @@ jobs:
           python -W ignore -m pip install -r requirements-all.txt
       - name: Build docs
         run: sphinx-build docs/source docs/build/html -W --keep-going -j auto
+      - name: Upload docs
+        uses: actions/upload-artifact@v3
+        with:
+          name: built docs
```
Comment (Bibo-Joshi) [inline .github/workflows/docs.yml:36]:
```suggestion
          name: HTML Docs
```

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -30,3 +30,12 @@ jobs:
...
+          retention-days: 7
+          path: |
```
Comment (Bibo-Joshi) [inline .github/workflows/docs.yml:38]: Can you add a short comment on what this does?

**Review by Bibo-Joshi — COMMENTED** [review]
Also a nice idea! TBH I guess the use case is not overly big, but neither is the effort 🙃

File: .github/workflows/docs.yml (hunk 2 above)
Comment (harshil21) [reply .github/workflows/docs.yml:38]: the days? or the path?

Comment (Bibo-Joshi) [reply .github/workflows/docs.yml:38]: The path

**Review by Bibo-Joshi — APPROVED** [review]
Thanks for the updates! As said above, even if the advantage may not be huge, I see no downsides of this change. I'll give the others a chance to chime in if they see the need and merge on thursday evening otherwise :)

**Pull request merged**